CMakeLists.txt

```diff
@@ -0,0 +1,256 @@
+cmake_minimum_required(VERSION 3.15.0)
```

Comment:
Why `3.15.0`. Just asking to understand rational @mathisloge

Reply:
I need to check the minimum supported version. Will try to get it as low as possible